packages/walletconnect-v2/src/index.ts

```diff
@@ -86,27 +102,54 @@ export class WalletConnect extends Connector {
     this.events.emit(URI_AVAILABLE, uri)
   }
 
+  private async initializeProvider(
+    desiredChainId: number | undefined = this.defaultChainId
+  ): Promise<WalletConnectProvider> {
+    const ethProviderModule = await import('@walletconnect/ethereum-provider')
+    const rpcMap = this.rpcMap ? await getBestUrlMap(this.rpcMap, this.timeout) : undefined
+    const chainProps = this.getChainProps(desiredChainId)
+
+    this.provider = await ethProviderModule.default.init({
+      ...this.options,
+      ...chainProps,
+      rpcMap,
+    })
+
+    return this.handleProviderEvents(this.provider)
+  }
+
+  // Helper function to reorder chains array based on desiredChainId
+  private reorderChainsBasedOnId(
+    chains: number[] | undefined,
+    desiredChainId: number | undefined
+  ): number[] | undefined {
+    if (!chains || !desiredChainId || !chains.includes(desiredChainId) || chains.length === 0) {
+      return chains
+    }
+    return getChainsWithDefault(chains, desiredChainId)
```

Comment:
This seems unnecessary to me. Why not include that `if` additional logic from lines 126-128 inside `getChainsWithDefault`? We spent a lot of time recently with @cmcewen thinking about the ways to name this function properly and while we debated `reoderChains`, we figured `getWithDefault` makes more sense to emphasise it's literally putting the first one as default.

Reply:
It's just a question of where we put the new type checks and changes. `getChainsWithDefault` does not support `chains` or `optionalChains` as arguments any longer with the new upstream `options` type changes from `@walletconnect/ethereum-provider`. We either need to diff the caller or diff the util. I chose the caller, but I don't have a preference between the two.

Reply:
Yeah, I understand your mixed feelings there. I think - personally - since the only utility of that `getChainsWithDefault` was to cope with WC "abstraction" of choosing the preferred chain, I think putting it all there would make sense.

Reply:
BTW, just one tiny consideration for the future work: we could consider (if we're putting chains and optional chains as a class property) to e.g. make their type "arrayoneormore | undefined" and do not set them if they're either not defined or empty in length. That way, I think typings would become much easier down the road.